src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -185,4 +191,66 @@
         r.assertBuildStatusSuccess(r.waitForCompletion(b4));
     }
 
+    @Issue("JENKINS-34005")
+    @Test public void triggersProperty() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        // Verify the base case behavior.
+        p.setDefinition(new CpsFlowDefinition("echo 'foo'"));
+
+        assertTrue(p.getTriggers().isEmpty());
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        // Make sure the triggers are still empty.
+        assertTrue(p.getTriggers().isEmpty());
+
+        // Now add a trigger.
+        p.setDefinition(new CpsFlowDefinition("properties([[$class: 'PipelineTriggersJobProperty',\n"
+                + "  triggers: [[$class: 'TimerTrigger', spec: '@daily']]]])\n"
+                + "echo 'foo'"));
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        assertEquals(1, p.getTriggers().size());
+
```

Comment:
:bug: you are not verifying that `start` was called with the correct argument.

> calling `startTriggers()` in `PipelineTriggersJobProperty.setOwner` causes...chaos. Way too many calls to `startTriggers()`

Yet this may be the only way you get that call done properly. Why would there be multiple calls to `setOwner`?